File: problems/001-anagram-detection/c/optimized.c
#define solve __glifex_ref_optimized
#include "solution.h"

JVal *solve(JVal *c) {
    const char *s = jget(c, "s")->str, *t = jget(c, "t")->str;
    if (strlen(s) != strlen(t)) return jbool_(0);
    int count[256] = {0};
    for (const unsigned char *p = (const unsigned char *)s; *p; p++) count[*p]++;
    for (const unsigned char *p = (const unsigned char *)t; *p; p++)
        if (--count[*p] < 0) return jbool_(0);
    return jbool_(1);
}

File: problems/001-anagram-detection/dart/optimized.dart
dynamic solve(Map<String, dynamic> c) {
  final s = c['s'] as String, t = c['t'] as String;
  if (s.length != t.length) return false;
  final count = <int, int>{};
  for (final u in s.codeUnits) count[u] = (count[u] ?? 0) + 1;
  for (final u in t.codeUnits) {
    final n = (count[u] ?? 0) - 1;
    if (n < 0) return false;
    count[u] = n;
  }
  return true;
}
